src/Magnum/Math/Geometry/Intersection.h

```diff
@@ -123,8 +125,74 @@ class Intersection {
             const T f = dot(planePosition, planeNormal);
             return (f-dot(planeNormal, p))/dot(planeNormal, r);
         }
+
+        /**
+         * @brief Intersection of a point and a camera frustum
+         * @param point Point
+         * @param frustum Frustum planes with normals pointing outwards
+         * @return `true` if the point is on or inside the frustum.
+         *
+         * @todoc formulas used
+         */
+        template<class T> bool pointFrustum(const Vector3<T>& point, const Vector4<T> (&frustum)[6]) {
+            for(const Vector4<T>& f : frustum) {
+                if(pointPlaneHalfSpace(point, f) < 0) {
+                    /* the point is in front of one of the frustum planes (normals point outwards) */
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        /**
+         * @brief Intersection of a range and a camera frustum
+         * @param box Range
+         * @param frustum Frustum planes with normals pointing outwards
+         * @return `true` if the box intersects with the camera frustum.
+         */
+        template<class T> bool rangeFrustum(const Range3D<T>& box, Containers::StaticArrayView<6, const Vector4<T>> frustum);
```

Comment:
I thought about this for quite some time -- from what I've seen on various places, there are currently multiple ways to represent the frustum planes:

- `Vector4[6]` (used above)
- `std::vector<Vector4>` (used in the shadows example)
- `std::array<Vector4, 6>`
- ...
- `Containers::StaticArrayView<6, Vector4>` (used in this function)

Neither of these is clearly a winner and converting the representations in order to make it possible to call a function that takes one particular kind is annoying, sometimes impossible. Because of that, I'd like to have a clearly defined `Math::Frustum` type for representing frustums in the engine instead of having a bunch of free functions and letting the user spend costly brain cycles reinventing the storage format every time:

```cpp
namespace Magnum { namespace Math {

template<class T> class Frustum {
    public:
        constexpr /*implicit*/ Frustum(const Vector4<T>& plane0, const Vector4<T>& plane1, const Vector4<T>& plane2, const Vector4<T>& plane3, const Vector4<T>& plane4, const Vector4<T>& plane5): _data{plane0, plane1, plane2, plane3, plane4, plane5} {}
        
        explicit Frustum(const Matrix4<T>& projectionMatrix): _data{...} {
            // I guess such conversion is already done somewhere? is an inverse operation possible?
        }
        
        T* data() { return _data[0].data(); }
        constexpr const T* data() const { return _data[0].data(); }
        
        constexpr Containers::StaticArrayView<6, Vector4<T>> planes() const { return _data; }
        
        constexpr Vector4<T> operator[](std::size_t i) const { return _data[i]; }

        Containers::StaticArray<6, Vector3<T>> corners() const {
            // maybe later? the code is in the shadows example
        }

    private:
        Vector4<T> _data[6];
};

}}
```

There is a bunch of "boilerplate" stuff missing (like debug operators, typedefs in root namespace, conversion constructors etc.), I can add that myself later.

Originally I wanted to put the point/box intersection routines inside the class, but after thinking about that a bit, I'm not really sure -- what do you think? Pros of having member function is that it's all very self-contained (one doesn't need to include another header), cons are that the class might explode later with stuff that not everybody needs (intersection of frustum and lines, AABBs, OBBs, spheres, intersecting stereo frustums with objects...). So I would keep them here I think.

Also, because I assume this functionality will get used quite a lot in quite tight loops, I imagine there could be some additions for batch processing (I guess you remember the [famous Mike Acton review](https://macton.smugmug.com/Other/2008-07-15-by-Eye-Fi/n-xmKDH/i-BrHWXdJ)). So for example a `cull()` function, working similarly to `std::remove()` (receiving mutable array view of boxes, reordering them, returning count of boxes that actually intersect the frustum) and then [optimizing the hell out of it](https://fgiesen.wordpress.com/2010/10/17/view-frustum-culling/). But... not *really* requesting that now, a TODO for later :)